lib/compat/wordpress-6.0/class-wp-theme-json-gutenberg.php

```diff
@@ -16,6 +16,50 @@
  */
 class WP_Theme_JSON_Gutenberg extends WP_Theme_JSON_5_9 {
 
+	/**
+	 * Metadata for style properties.
+	 *
+	 * Each element is a direct mapping from the CSS property name to the
+	 * path to the value in theme.json & block attributes.
+	 */
+	const PROPERTIES_METADATA = array(
+		'background'                        => array( 'color', 'gradient' ),
+		'background-color'                  => array( 'color', 'background' ),
+		'border-radius'                     => array( 'border', 'radius' ),
+		'border-top-left-radius'            => array( 'border', 'radius', 'topLeft' ),
+		'border-top-right-radius'           => array( 'border', 'radius', 'topRight' ),
+		'border-bottom-left-radius'         => array( 'border', 'radius', 'bottomLeft' ),
+		'border-bottom-right-radius'        => array( 'border', 'radius', 'bottomRight' ),
+		'border-color'                      => array( 'border', 'color' ),
+		'border-width'                      => array( 'border', 'width' ),
+		'border-style'                      => array( 'border', 'style' ),
+		'color'                             => array( 'color', 'text' ),
+		'font-family'                       => array( 'typography', 'fontFamily' ),
+		'font-size'                         => array( 'typography', 'fontSize' ),
+		'font-style'                        => array( 'typography', 'fontStyle' ),
+		'font-weight'                       => array( 'typography', 'fontWeight' ),
+		'letter-spacing'                    => array( 'typography', 'letterSpacing' ),
+		'line-height'                       => array( 'typography', 'lineHeight' ),
+		'margin'                            => array( 'spacing', 'margin' ),
+		'margin-top'                        => array( 'spacing', 'margin', 'top' ),
+		'margin-right'                      => array( 'spacing', 'margin', 'right' ),
+		'margin-bottom'                     => array( 'spacing', 'margin', 'bottom' ),
+		'margin-left'                       => array( 'spacing', 'margin', 'left' ),
+		'padding'                           => array( 'spacing', 'padding' ),
```

Comment:
Should we give the same treatment to 'padding'?

If I add a padding value to theme.json, 

```json
                "spacing": {
                        "blockGap": "1.5rem",
                        "padding": "88px"
                },
```

I see the following:

```css
body {
	padding: 88px;
	--wp--style--block-gap: 1.5rem;
}
```

Not sure what the intention is here, but If there were another CSS var for a single padding value, e.g., `--wp--style--root--padding`, we could use it as a fallback, e.g., 

```css
padding-top: var(--wp--style--root--padding-top, var(--wp--style--root--padding))
```

Reply:
We won't be able to use `--wp--style--root--padding` directly as a fallback though, because it might be a multi-value format e.g. `32px 42px`. Not quite sure how best to handle that - is that something that we've solved for gap already?

Reply:
> We won't be able to use --wp--style--root--padding directly as a fallback though, because it might be a multi-value format e.g. 32px 42px. 

Oh, excellent point. 😆 Good catch.

I'm not sure splitting by `' '` would be robust enough (I tried in an old gap commit and was told the same). 

As for gap, the gap value can be an array, so we check for that and split the values accordingly. 

See: https://github.com/WordPress/gutenberg/blob/trunk/lib/block-supports/layout.php#L70

Otherwise we assume string.

Reply:
Hmmm, here we'll always have a string so that won't work. 

If we could identify which of the variables are set, we could maybe output different styles accordingly, but I'm not sure how that would work on the editor side (the whole editor side of this PR is still quite iffy though, so maybe a solution will present itself in time)

Reply:
Thinking better of this, we always need to know the specific values for left and right padding because we need them for the negative margins on full width blocks. For that we need to separate the multi-value padding string into its component parts, so I think we'll have to do it the messy way.

Reply:
I updated the code below to separate out the values from a shorthand padding declaration. This currently only works on the server side; looking into how to implement it for the editor now.